dev/tasks/python-wheels/github.osx.arm64.yml

```diff
@@ -134,6 +134,8 @@ jobs:
           $PYTHON -m venv test-arm64-env
           source test-arm64-env/bin/activate
           pip install --upgrade pip wheel
+          arch -arm64 pip install -r arrow/python/requirements-wheel-test.txt
```

Comment:
Is the `arch` call necessary here? `pip` should point to the `test-arm64-env` pip install which already should have the right architecture... (same below for x86)

Reply:
Why the `$PYTHON` (`/Library/Frameworks/Python.framework/Versions/3.10/bin/python3.10`) is a x86 Python.
If we don't use `arch -arm64`, wheels for x86 are installed.

See https://github.com/ursacomputing/crossbow/runs/6165284883?check_suite_focus=true#step:10:147 for example:

```text
  Using cached MarkupSafe-2.1.1-cp310-cp310-macosx_10_9_x86_64.whl (13 kB)
```

We can omit `arch -x86_64` in "Test Wheel on AMD64" but I think that we should specify `arch -x86_64` explicitly to work on macOS that provides arm64 system Python too.

Reply:
I don't have M1 mac. I hope that someone confirms system Python on M1 mac and `arch` relations...

Reply:
Also the [following question + answer](https://github.com/numpy/numpy/issues/18143#issuecomment-759389439) seems relevant:

> > will the universal2 python.org installer pick up thin arm64 wheels if they are available and Python is started in native mode?
> That's more a pip question than python version, but yes pip will use arm64 wheels when running in natively on M1 Macs.

So even a universal Python should install arm64 wheels by default (if not run specifically in x86 mode).

Reply:
Ok, I understand now. Our self-hosted GHA runner is probably a x86 binary, so runs child processes in x86 mode by default.